Resources/Prototypes/Entities/Objects/Weapons/Throwable/canister_grenades.yml

```
@@ -0,0 +1,117 @@
+- type: entity
+  parent: [VolatileGrenadeBase, GrenadeBase, BaseSecurityContraband ]
+  id: SmokeGrenade
+  name: smoke grenade
+  description: A tactical grenade that releases a large, long-lasting cloud of smoke when used.
+  components:
+  - type: Sprite
+    sprite: Objects/Weapons/Grenades/smoke.rsi
+  - type: SmokeOnTrigger
+    keysIn:
+    - timer
+    duration: 30
+    spreadAmount: 50
+  - type: EmitSoundOnTrigger
+    keysIn:
+    - timer
```

This is a yaml trick  called anchors and aliases for condensing lines and setting constants. Use the & on the first instance, then the * on every subsequent `keysIn:` in the file. It's scoped per-file.

```suggestion
  - type: SmokeOnTrigger
    keysIn: &TimerIn
    - timer
    duration: 30
    spreadAmount: 50
  - type: EmitSoundOnTrigger
    keysIn: *TimerIn
```